src/player/process-death.cpp

```diff
@@ -42,172 +42,155 @@
  * Centers a string within a GRAVE_LINE_WIDTH character string		-JWT-
  * @details
  */
-static void center_string(char *buf, concptr str)
+static std::string center_string(const std::string &str)
 {
-    int i = strlen(str);
-    int j = GRAVE_LINE_WIDTH / 2 - i / 2;
-    (void)sprintf(buf, "%*s%s%*s", j, "", str, GRAVE_LINE_WIDTH - i - j, "");
+    int j = GRAVE_LINE_WIDTH / 2 - str.length() / 2;
+    return std::string().append(j, ' ').append(str).append(GRAVE_LINE_WIDTH - str.length() - j, ' ');
 }
 
 /*!
  * @brief 墓に基本情報を表示
  * @param player_ptr プレイヤーへの参照ポインタ
- * @param buf 墓テンプレ
  */
-static void show_basic_params(PlayerType *player_ptr, char *buf)
+static void show_basic_params(PlayerType *player_ptr)
 {
-    char tomb_message[160];
-    (void)sprintf(tomb_message, _("レベル: %d", "Level: %d"), (int)player_ptr->lev);
-    center_string(buf, tomb_message);
-    put_str(buf, 11, 11);
-
-    (void)sprintf(tomb_message, _("経験値: %ld", "Exp: %ld"), (long)player_ptr->exp);
-    center_string(buf, tomb_message);
-    put_str(buf, 12, 11);
-
-    (void)sprintf(tomb_message, _("所持金: %ld", "AU: %ld"), (long)player_ptr->au);
-    center_string(buf, tomb_message);
-    put_str(buf, 13, 11);
+    put_str(center_string(format(_("レベル: %d", "Level: %d"), (int)player_ptr->lev)).data(), 11, 11);
+
+    put_str(center_string(format(_("経験値: %ld", "Exp: %ld"), (long)player_ptr->exp)).data(), 12, 11);
+
+    put_str(center_string(format(_("所持金: %ld", "AU: %ld"), (long)player_ptr->au)).data(), 13, 11);
 }
 
 #ifdef JP
 /*!
  * @brief プレイヤーを殺したモンスターを表示する (日本語版専用)
  * @param player_ptr プレイヤーへの参照ポインタ
- * @param buf 墓テンプレ
- * @param tomb_message 墓碑に刻む言葉
- * @return 追加の行数
+ * @return メッセージと余分な行数のペア
  */
-static int show_killing_monster(PlayerType *player_ptr, char *buf, char *tomb_message, size_t tomb_message_size)
+static std::pair<std::string, int> show_killing_monster(PlayerType *player_ptr)
 {
-    shape_buffer(player_ptr->died_from.data(), GRAVE_LINE_WIDTH + 1, tomb_message, tomb_message_size);
-    char *t;
-    t = tomb_message + strlen(tomb_message) + 1;
+    char buf[160];
+    shape_buffer(player_ptr->died_from.data(), GRAVE_LINE_WIDTH + 1, buf, sizeof(buf));
+    char *t = buf + strlen(buf) + 1;
     if (!*t) {
-        return 0;
+        return std::pair<std::string, int>(buf, 0);
```

Comment:
You can use `std::make_pair(buf, 0);` instead.
This eliminates the need to explicitly specify the type and makes the process slightly more efficient.
The same is true at the end of this function.

Reply:
The forced push change those two returns to use make_pair().  It also makes use of it in show_tomb_detail().